packages/runtime/src/api/templates.ts

```diff
@@ -0,0 +1,102 @@
+import { getCopyFunction } from '../prop-controllers/copy'
+import { ReactRuntime } from '../runtimes/react'
+import {
+  getComponentPropControllerDescriptors,
+  getPropControllerDescriptors,
+  State,
+} from '../state/react-page'
+
+type ElementData = {
+  type: string
+  key: string
+  props: any
```

Comment:
Note - there is **a lot** of types I could bring into the runtime.

I want some, but I don't want to go overboard. This is where I drew that line, having `props` be of type `any`. I believe this is the right call, because for code components, we have no type information.

Reply:
Most types should already be defined in the runtime